v2/pkg/genruntime/extensions/error_classifier.go

```diff
@@ -42,11 +42,18 @@ func CreateErrorClassifier(
 	}
 
 	return func(cloudError *genericarmclient.CloudError) (core.CloudErrorDetails, error) {
-		log.V(Info).Info(
-			"Classifying cloud error",
-			"Message", cloudError.InnerError.Message,
-			"Code", cloudError.InnerError.Code,
-			"Target", cloudError.InnerError.Target)
+		if cloudError.InnerError != nil {
+			log.V(Info).Info(
+				"Classifying cloud error",
+				"Error", cloudError.Error(),
+				"Message", cloudError.InnerError.Message,
+				"Code", cloudError.InnerError.Code,
+				"Target", cloudError.InnerError.Target)
+		} else {
+			log.V(Info).Info(
+				"Classifying cloud error",
+				"Error", cloudError.Error())
+		}
 
 		result, err := impl.ClassifyError(cloudError, apiVersion, log, classifier)
 
```

Comment:
minor: I can't actually comment below here but I think you might want to check `err != nil` and either log a separate log for that, or just return before the log below, as if you don't you'll log "Cloud error classified Classification= "", Code="",  Message=""" which seems like it has a high probability of being confusing (since actually the error was _not_ classified and there was an error).

Reply:
Good idea. Done.